csrc/id_model/id_model.cpp

```diff
@@ -540,6 +558,41 @@ void IdModel::buildPermissiveGraph() {
   idGraph(IdMappingMode::PERMISSIVE).validateConsistency();
 }
 
+namespace {
+
+// Returns the root producer iteration domains that are resolved by provided
+// consumer
+std::unordered_map<IterDomain*, IterDomain*> resolvedRootBroadcasts(
```

Comment:
If the only use for this map is to iterate through it, should we instead use a `std::vector<std::pair<IterDomain*, IterDomain*>>` which is both faster and deterministic?